src/esp/assets/ResourceManager.cpp

```diff
@@ -2238,15 +2563,188 @@ void ResourceManager::loadMaterials(Importer& importer,
                 iMaterial, assetName));
       }
+
+      // Merge all custom attribute except remapped texture pointers with
+      // original material for final material. custMaterialData should never be
+      // Cr::Containers::NullOpt since every non-error branch is covered.
+      Cr::Containers::Optional<Mn::Trade::MaterialData> mergedCustomMaterial =
+          Mn::MaterialTools::merge(
+              *custMaterialData, *materialData,
+              Mn::MaterialTools::MergeConflicts::KeepFirstIfSameType);
+
+      // Now build texture pointer array, with appropriate layers based on
+      // number of layers in original material
+
+      // New txtrptr-holding material's attributes
+      Cr::Containers::Array<Mn::Trade::MaterialAttributeData> newAttributes{};
+      // New txtrptr-holding material's layers
+      Cr::Containers::Array<Mn::UnsignedInt> newLayers{};
+
+      // Copy all texture pointers into array
+      // list of all this material's attributes
+      const Cr::Containers::ArrayView<const Mn::Trade::MaterialAttributeData>
+          materialAttributes = materialData->attributeData();
+      // Returns nullptr if has no attributes
+      if (materialAttributes != nullptr) {
+        // For all layers
+        for (int layerIdx = 0; layerIdx < numMaterialLayers; ++layerIdx) {
+          // Add all material texture pointers into new attributes
+          // find start and end idxs for each layer
+          int stIdx = materialData->attributeDataOffset(layerIdx);
+          int endIdx = materialData->attributeDataOffset(layerIdx + 1);
+
+          for (int mIdx = stIdx; mIdx < endIdx; ++mIdx) {
+            const Mn::Trade::MaterialAttributeData& materialAttribute =
+                materialAttributes[mIdx];
+            auto attrName = materialAttribute.name();
+            const auto matType = materialAttribute.type();
+            // Find textures and add them to newAttributes
+            // bool found = (std::string::npos != key.find(strToLookFor));
+            if ((matType == Mn::Trade::MaterialAttributeType::UnsignedInt) &&
+                attrName.hasSuffix("Texture")) {
+              // texture index, copy texture pointer into newAttributes
+              const Mn::UnsignedInt txtrIdx =
+                  materialAttribute.value<Mn::UnsignedInt>();
+              // copy texture into new attributes tagged with lowercase material
+              // name
+              auto newAttrName = Cr::Utility::formatString(
+                  "{}{}Ptr",
+                  Cr::Utility::String::lowercase(attrName.slice(0, 1)),
+                  attrName.slice(1, attrName.size()));
+              // Temporary debug message
+              Cr::Utility::formatInto(debugStr, debugStr.size(),
+                                      "| txtr ptr name:{} | idx :{}  Layer {}",
+                                      newAttrName, (textureBaseIndex + txtrIdx),
+                                      layerIdx);
```

Comment:
Do we want to retain this debug message?

Reply:
It is being sent to the debug stream, so it can be easily disabled.  It's certainly more informative than the current debug message to which it appends.

Reply:
Sounds good! I would remove the `Temporary debug message` comment so that it doesn't get removed by somebody else.